docs/src/components/power-estimate.js

```diff
@@ -0,0 +1,240 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { displayPower, underglowPower, zmkBase } from "../data/power";
+import "../css/power-estimate.css";
+
+const lithiumIonMonthlyDischargePercent = 5;
+const lithiumIonAverageVoltage = 3.8;
+const lithiumIonDischargeEfficiency = 0.85;
+const lithiumIonDischargeEffiecincyRange = 0.05;
+
+const timeSpentTyping = 0.02;
+
+const measurementAccuracy = 0.2; // Nordic power profiler kit accuracy
+
+const batVolt = lithiumIonAverageVoltage;
+
+const palette = [
+  "#bbdefb",
+  "#90caf9",
+  "#64b5f6",
+  "#42a5f5",
+  "#2196f3",
+  "#1e88e5",
+  "#1976d2",
+];
+
+function formatUsage(usage) {
+  if (usage > 1000) {
+    return (usage / 1000).toFixed(1) + "mW";
+  }
+
+  return Math.round(usage) + "µW";
+}
+
+function voltageEquivalentCalc(powerSupply) {
+  if (powerSupply.type === "LDO") {
+    return batVolt;
+  } else if (powerSupply.type === "SWITCHING") {
+    return powerSupply.outputVoltage / powerSupply.efficiency;
+  }
+}
+
+function formatMinutes(minutes) {
+  let message = "";
+  let count = 0;
+
+  let units = ["year", "month", "week", "day", "minute"];
+  let multiples = [60 * 24 * 365, 60 * 24 * 30, 60 * 24 * 7, 60 * 24, 60, 1];
+
+  for (let i = 0; i < units.length; i++) {
+    if (minutes >= multiples[i]) {
+      const timeCount = Math.floor(minutes / multiples[i]);
+      minutes -= timeCount * multiples[i];
+      count++;
+      message +=
+        timeCount + (timeCount > 1 ? ` ${units[i]}s ` : ` ${units[i]} `);
+    }
+
+    if (count == 2) return message;
+  }
+
+  return message || "0 minutes";
+}
+
+function PowerEstimate({
+  board,
+  splitType,
+  batterymAh,
+  usage,
+  underglow,
+  display,
+}) {
+  if (!board || !board.powerSupply.type || !batterymAh) {
+    return (
+      <div className="powerEstimate">
+        <h3>About ? {splitType !== "standalone" ? "on " + splitType : ""}</h3>
+        <div className="powerEstimateBar">
+          <div
+            className="powerEstimateBarEl"
+            style={{
+              width: "100%",
+              background: "#e0e0e0",
+              mixBlendMode: "overlay",
+            }}
+          ></div>
+        </div>
+      </div>
+    );
+  }
+
+  const powerUsage = [];
+  let totalUsage = 0;
+
+  const voltageEquivalent = voltageEquivalentCalc(board.powerSupply);
+
+  // Lithium ion self discharge
+  const lithiumDischarge =
+    ((parseInt(batterymAh) * 1000 * lithiumIonMonthlyDischargePercent) /
```

Comment:
I've redone how these are all calculated so that the variables are much more direct and found the only real calculation I can pull out is the conversion from micro amps to milliwatt usage, which feels superfluous. Hopefully what I've updated to is much more readable (should only need to know ohms law and metric conversion now 😅).